cluster/log-dump/log-dump.sh

```diff
@@ -232,7 +232,7 @@ function dump_nodes() {
   else
     echo "Detecting nodes in the cluster"
     detect-node-names &> /dev/null
-    node_names=( "${NODE_NAMES[@]}" )
+    node_names=( "${NODE_NAMES[@]:+${NODE_NAMES[@]}}" )
```

Comment:
This is making the script hard to understand. Instead of changes like this here and below, I guess you can simply do the following:

```
detect-node-names &> /dev/null
if [[ -n "${NODE_NAMES:-}" ]]; then
  node_names=( "${NODE_NAMES[@]}" )
fi
```

The code further should handle the case that `node_names` is empty.